### PR TITLE
Add low-level SDL AST helpers

### LINKED ISSUE #9: Add low-level SDL AST helpers
## Parent

#7

## Slice type

AFK

## What to build

Add small AST-level helpers that are useful even without a full schema index: render wrapped type references, unwrap them to the named type, and read const string directive arguments such as `@deprecated(reason:)`.

## Acceptance criteria

- [ ] A helper renders nested type references exactly, including cases such as `[Book!]!`.
- [ ] A helper unwraps nested list/non-null references to the named type, such as `Book` from `[Book!]!`.
- [ ] A directive helper reads a string argument from a named directive and reports absence or non-string values predictably.
- [ ] Tests cover `@deprecated` with a `reason`, `@deprecated` without a `reason`, missing directives, and nested type wrappers.

## Blocked by

None - can start immediately

## User stories addressed

- Derived user story US6: Callers can render and unwrap GraphQL type references.
- Derived user story US7: Callers can read const string directive args like `@deprecated(reason:)`.

### PR DESCRIPTION
**Pull request by brian-bell**
## Summary

- Add `ast.TypeString` for canonical rendering of GraphQL type references such as `[Book!]!`.
- Add `ast.NamedTypeName` to unwrap list and non-null wrappers to the innermost named type.
- Add `ast.DirectiveStringArg` for reading decoded string arguments from named directives, including predictable false results for missing and non-string values.
- Document the helper APIs in the README and cover edge cases with AST package tests.

Closes #9

## Verification

- `gofmt -l .`
- `go test ./...`
- `go vet ./...`
- `git diff --check origin/main...HEAD`

### CODE REVIEW

File: ast/document.go [hunk 1]
```diff
@@ -43,6 +43,30 @@ func (dl DirectiveList) ForName(name string) *Directive {
...
+			if arg == nil || arg.Name != argumentName {
+				continue
+			}
+			value, ok := arg.Value.(*StringValue)
+			if !ok || value == nil {
+				return "", false
+			}
+			return value.Value, true
+		}
+		return "", false
```
[the remark this answers is not in the document]
Comment (brian-bell) [reply ast/document.go]: Fixed in 1814af7.

`DirectiveStringArg` now continues scanning later repeated directives when an earlier matching directive does not contain the requested argument. It still returns `false` when the first matching argument is present but not a string, preserving the existing non-string behavior.

Added `TestDirectiveStringArg_ScansRepeatableDirectives` for the `@tag @tag(name: "later")` case.

Verification:
- `go test ./ast -run TestDirectiveStringArg`
- `gofmt -l .`
- `go test ./...`
- `go vet ./...`
- `git diff --check`
- `~/.skill-symlinks/skills/autoreview/scripts/autoreview --mode local --prompt ...` clean: no accepted/actionable findings
- PR CI passed: gofmt, ubuntu test, macOS test, fuzz, benchmarks
